projects/schematics/src/wrapper-module/index.ts

```diff
@@ -0,0 +1,563 @@
+ * E.g. for the given `(m) => m.CheckoutModule`
+ * it replaces it with the given module name: `(m) => m.CheckoutWrapperModule`.
+ */
+function updateDynamicImportModuleName(
+  dynamicImport: ArrowFunction,
+  wrapperModuleName: string
+): void {
+  getDynamicImportPropertyAccess(dynamicImport)?.replaceWithText(
+    `m.${wrapperModuleName}`
+  );
+}
+
+function orderWrapperFeatures(
+  options: SpartacusWrapperOptions,
+  markerFeatureModulePath: string
+): Rule {
+  return (tree: Tree, context: SchematicContext) => {
+    if (options.debug) {
+      context.logger.info(
+        formatFeatureStart(
+          markerFeatureModulePath,
+          `ordering features in the wrapper module...`
+        )
+      );
+    }
+
+    const basePath = process.cwd();
+    const { buildPaths } = getProjectTsConfigPaths(tree, options.project);
+
+    for (const tsconfigPath of buildPaths) {
+      const { appSourceFiles } = createProgram(tree, basePath, tsconfigPath);
+
+      for (const wrapperModule of appSourceFiles) {
+        if (!wrapperModule.getFilePath().includes(markerFeatureModulePath)) {
+          continue;
+        }
+
+        const analysis = analyzeFeature(wrapperModule);
+        if (analysis.unrecognized?.length) {
+          return noop();
+        }
+
+        const ordered = orderFeatures(analysis);
+        getModulePropertyInitializer(
+          wrapperModule,
+          'imports',
+          false
+        )?.replaceWithText(`[${ordered.join(',\n')}]`);
+
+        saveAndFormat(wrapperModule);
+        break;
+      }
+    }
+
+    if (options.debug) {
+      context.logger.info(
+        formatFeatureComplete(
+          markerFeatureModulePath,
+          `features ordered in the wrapper module.`
+        )
+      );
+    }
+  };
+}
+
+/**
+ * Generates wrapper modules for the given
+ * Spartacus feature module.
+ */
+export function generateWrapperModule(options: SpartacusWrapperOptions): Rule {
+  return (_tree: Tree, _context: SchematicContext): Rule => {
+    const markerFeatureConfig = getSchematicsConfigByFeatureOrThrow(
+      getKeyByMappingValueOrThrow(
+        featureFeatureModuleMapping,
+        options.markerModuleName
+      )
+    );
+    const folderPath = createSpartacusFeatureFolderPath(
+      markerFeatureConfig.folderName
+    );
+    const fileName = createSpartacusWrapperModuleFileName(
+      markerFeatureConfig.moduleName
+    );
+    const markerFeatureModulePath = `${folderPath}/${fileName}`;
+
+    return chain([
+      createWrapperModule({
+        project: options.project,
+        moduleName: options.markerModuleName,
+        debug: options.debug,
+      }),
+
+      updateWrapperModule({
+        project: options.project,
+        moduleName: options.markerModuleName,
+        markerFeatureModulePath,
+        debug: options.debug,
+      }),
+      updateWrapperModule({
+        project: options.project,
+        moduleName: options.featureModuleName,
+        markerFeatureModulePath,
```

Comment:
Kind of confused when I'm trying to understand

why do we have to updateWrapperModule twice for the marker module name and the feature module that will be added to the wrapper

Reply:
Because we need to statically import both the marker (`markerModuleName`) and target feature module (`featureModuleName`) to the wrapper module.